runPerftool.psm1

```diff
@@ -286,44 +286,342 @@ Function ProcessCommands{
+
+    [Parameter(Mandatory=$True, ParameterSetName="Default", HelpMessage="Dest Machine Password?")]
+    [Parameter(Mandatory=$True, ParameterSetName="LocalTransmit", HelpMessage="Dest Machine Password?")]
+    [Parameter(Mandatory=$True, ParameterSetName="RemoteTransmit", HelpMessage="Dest Machine Password?")]
+    [SecureString] $DestIpPassword,
+
+    [Parameter(Mandatory=$True, HelpMessage="Src Machine Username?")]
+    [string] $SrcIpUserName,
 
-    [pscredential] $transmitIPCreds = $null 
+    [Parameter(Mandatory=$True, ParameterSetName="Default", HelpMessage="Src Machine Password?")]
+    [Parameter(Mandatory=$True, ParameterSetName="LocalTransmit", HelpMessage="Src Machine Password?")]
+    [Parameter(Mandatory=$True, ParameterSetName="RemoteTransmit", HelpMessage="Src Machine Password?")]
+    [SecureString] $SrcIpPassword,
 
-    if ($TransmitEventsRemotely) {
-        [pscredential] $transmitIPCreds = New-Object System.Management.Automation.PSCredential($TransmitUserName, $TransmitPassword)
+    [Parameter(Mandatory=$True)]
+    [string] $CommandsDir,
+
+    [Parameter(Mandatory=$False)]
+    [string] $Bcleanup=$True,
+
+    [Parameter(Mandatory=$False)]
+    [Boolean] $ZipResults=$True,
+
+    [Parameter(Mandatory=$False)]
+    [Int] $TimeoutValueInSeconds=90,
+
+    [Parameter(Mandatory=$False)]
+    [Int] $PollTimeInSeconds=5,
+
+    [Parameter(Mandatory=$True, ParameterSetName="LocalTransmit")]
+    [Parameter(Mandatory=$True, ParameterSetName="LocalTransmitOnContainers")]
+    [Switch] $TransmitEventsLocally,
+
+    [Parameter(Mandatory=$True, ParameterSetName="RemoteTransmit")]
+    [Parameter(Mandatory=$True, ParameterSetName="RemoteTransmitOnContainers")]
+    [Switch] $TransmitEventsRemotely,
+
+    [Parameter(Mandatory=$True, ParameterSetName="RemoteTransmit")]
+    [Parameter(Mandatory=$True, ParameterSetName="RemoteTransmitOnContainers")]
+    [String] $TransmitIP,
+
+    [Parameter(Mandatory=$True, ParameterSetName="RemoteTransmit")]
+    [Parameter(Mandatory=$True, ParameterSetName="RemoteTransmitOnContainers")]
+    [String] $TransmitUserName,
+
+    [Parameter(Mandatory=$True, ParameterSetName="RemoteTransmit")]
+    [SecureString] $TransmitPassword,
+
+    [Parameter(Mandatory=$True, ParameterSetName="RunOnContainers")]
+    [Parameter(Mandatory=$True, ParameterSetName="LocalTransmitOnContainers")]
+    [Parameter(Mandatory=$True, ParameterSetName="RemoteTransmitOnContainers")]
+    [Switch] $RunOnContainers=$False,
+
+    [Parameter(Mandatory=$False, ParameterSetName="Default")]
+    [Parameter(Mandatory=$False, ParameterSetName="LocalTransmit")]
+    [Parameter(Mandatory=$False, ParameterSetName="RemoteTransmit")]
+    [Parameter(Mandatory=$True, ParameterSetName="RunOnContainers")]
+    [Parameter(Mandatory=$True, ParameterSetName="LocalTransmitOnContainers")]
+    [Parameter(Mandatory=$True, ParameterSetName="RemoteTransmitOnContainers")]
+    [Switch] $RunWithSinglePSSession=$False
+    )
```

Comment:
I'm a little confused here, especially about the LocalTransmit parameter. I think we should make as little change as possible to these script. I also think we should generalize the runOnContainers to runWithSSH since this is mainly adding SSH support if other people want to use it.

Reply:
Ok. I've updated the script to leave the original functions unchanged and added new functions that can handle both the old and new behaviors. So anyone using the old functions should be completely safe now.

I also replaced RunOnContainers with RunOverSSH and DisableFirewallConfiguration, which is the two things RunOnContainers was doing.